Obtain correct origin for the geometric image analysis

LucVV opened the pull request:
<!--
  Thanks a lot for contributing to our project!
  Please, do not remove any text from this template (unless instructed otherwise).
-->

## Proposed change
<!--
  What did you change and why did you change it?
-->
As discussed on the [OpticStudio forum](https://community.zemax.com/zos-api-12/incorrect-datagrid-miny-for-geometric-image-analysis-5426), the origin of the GeometricImageAnalysis is incorrect. It was suggested to calculate the origin from the reference, which is implemented in this PR. 

There is still a question that needs to be answered: when `wavelength` is set to "All",  `reference` to anaything but "PrimaryChief" and there are multiple wavelengths, is the OpticStudio reference the average of the references calculated with each wavelength? This question is asked on the OpticStudio forum as the manual does not provide an answer.

I have currently implemented this fix for the old version of the Geometric Image Analysis, especially since it is still to be converted to the new format. 

Due to the above reasons, this PR is set as draft.
 
## Type of change
<!--
  What type of change does your PR introduce to ZOSPy?
-->

- [ ] Example (a notebook demonstrating how to use ZOSPy for a specific application)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New analysis (a wrapper around an OpticStudio analysis)
- [ ] New feature (other than an analysis)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests
- [ ] Documentation (improvement of either the docstrings or the documentation website)

# Additional information
<!--
  We would like to know which versions of Python and OpticStudio you are running.
  This helps us to keep the compatibility section in our documentation updated.
  
  Please list all Python versions you used to test your changes. The unit tests will
  automatically try to test all currently supported Python versions. If you would like
  to do us a favor, install all necessary Python versions on your system. This helps
  us to ensure compatibility and detect any problems we might not be able to detect
  on our own systems. This makes your contribution even more valuable!
-->

- Python version: 3.11
- OpticStudio version: 20.3.2

## Related issues
<!--
  Please list any issues, discussions or pull requests related to this pull request.
-->

## Checklist
<!--
  Tick all boxes that apply. 
-->

- [x] I have followed the [contribution guidelines][contribution-guidelines]
- [x] The code has been linted, formatted and tested locally using tox.
- [ ] Local tests pass. **Please fix any problems before opening a PR. If this is not possible, specify what doesn't work and why you can't fix it.**
- [ ] I added new tests for any features contributed, or updated existing tests.
- [ ] I updated CHANGELOG.md with my changes (except for refactorings and changes in the documentation).

If you contributed an analysis:

- [ ] I did not use `AttrDict`s for the analysis result data (please use dataclasses instead).

If you contributed an example:

- [ ] I contributed my example as a Jupyter notebook.
    <!--
    This is important, because it allows users to see the results without
    executing the complete example.
    -->

<!--
  Thanks again for your contribution! We will look into it soon.
  Meanwhile, here are some useful resources that will help you to improve
  the quality of your contribution:
-->
[contribution-guidelines]: https://github.com/MREYE-LUMC/ZOSPy/blob/main/CONTRIBUTING.md
[unittest-instructions]: https://github.com/MREYE-LUMC/ZOSPy/blob/main/tests/README.md
[numpydoc]: https://numpydoc.readthedocs.io/en/latest/format.html

crnh commented:
This seems to be pretty complicated, I think it's best to add some unit tests for this as well...

LucVV commented:
I will add them soon!

crnh commented:
This should still be implemented for the new-style analysis!

crnh commented:
Two comments:

1. `_gia_correct_datagrid_origin` does not seem to be used anywhere;
2. In my opinion, using a wrapper class to fix this problem is unnecessarily complex. Instead, modifying the `unpack_datagrid` function to allow for optionally changing the minimum x and y values would work equally well and be a lot easier to implement, test and maintain.

LucVV commented:
This fix was actually much easier than expected. The origin of the datagrid is `-Settings.ImageSize/2`. I have implemented this for both `MinX` and `MinY` and added a test for it. The only thing I am unsure about is what happens to the output when `image_size` is lower than 0. The OpticStudio user interface seems to change its output based on what you select as `ShowAs`. I am not sure if we should limit `image_size` to >=0, or do something else. @crnh do you have an opinion on this?

crnh commented:
> I am not sure if we should limit `image_size` to >=0, or do something else. @crnh do you have an opinion on this?

As the behavior for negative image sizes is rather weird / unpredictable, I would be okay with not supporting negative image sizes. If later on they appear to be useful, it's easy enough to add support for them.

crnh commented on zospy/analyses/old/extendedscene.py:
```diff
@@ -8,9 +8,133 @@
+from zospy.api import constants, _ZOSAPI
 from zospy.zpcore import OpticStudioSystem
 
+class _GiaDataGridWrapper:
```
Is this wrapper still necessary? If not, I think it can be removed.

LucVV replied:
You are correct, I removed all changes from `analyses/old/extendescene`

crnh reviewed (changes requested):
The test reference data still needs to be updated. I'll do that for OpticStudio 2024.

crnh commented:
I ran the tests for OpticStudio 2024 R1 and R2, and they pass!

crnh reviewed (approved):
Nice work! Thanks for investigating and fixing this. For future reference, this behavior has been discussed on the OpticStudio forums: https://community.zemax.com/zos-api-12/incorrect-datagrid-miny-for-geometric-image-analysis-5426?postid=17836#post17836
